src/v/cluster/rm_stm.cc

```diff
@@ -1792,8 +1805,11 @@ ss::future<> rm_stm::apply_control(
 
         _mem_state.forget(pid);
 
-        if (_log_state.aborted.size() > _abort_index_segment_size) {
-            co_await make_snapshot();
+        if (
+          _log_state.aborted.size() > _abort_index_segment_size
+          && !_is_abort_idx_reduction_requested) {
+            ssx::spawn_with_gate(
+              _gate, [this] { return reduce_aborted_list(); });
```

Comment:
why is this in the background now? from what i can tell there isn't any sort of deadlock scenario, but maybe i'm missing something.

Reply:
I've added a comment [there](https://github.com/redpanda-data/redpanda/pull/5556/commits/adcc4406497d0f4b8d05a2cf3e12dcbeee5ae1e9#diff-31c312a88efb2dcf54d8fc5447d59316554d9f7df36fa3125cacbf768a3bdeb0R2091)